### PR TITLE
Quick Start for Existing Users V2: Assign new tour only to new existing users

### PR DESCRIPTION
**Pull request by ashiagr**
Parent #16350

This PR adds common tasks: `check_stats`, `view_site`, `follow_site` to a new `Quick Start` card and displays it only to new existing users.

Corresponding FluxC PR: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2387

https://user-images.githubusercontent.com/1405144/168042785-5c87a899-2866-43f6-bdfe-eb2d21cfed56.mov

To test: 

/cc @jostnes, @osullivanchris 

#### Test.1 Login Epilogue - `QuickStartForExistingUsersV2FeatureConfig` disabled - Displays old `Quick Start` card

1. Launch the app.
2. Logout/login and select "Show me around" when the quick start prompt is shown.
3. Select an existing site.
4. ✅ Notice that the old Quick Start card with `Customize` and `Grow` task types is shown.
5. Logout/login and select "Show me around" when the quick start prompt is shown.
6. Create a new site.
7. ✅ Notice that the old Quick Start card with `Customize` and `Grow` task types is shown.

#### Test.2 Site Picker - `QuickStartForExistingUsersV2FeatureConfig` disabled - Displays old `Quick Start` card

1. Continue from Test.1
2. Create a new site using the site picker
3. ✅ Notice that the old Quick Start card with `Customize` and `Grow` task types is shown

#### Test.3 Login Epilogue - `QuickStartForExistingUsersV2FeatureConfig` enabled - Displays new `Quick Start` card when site is selected and old `Quick Start` card when site is created

1. Launch the app
2. Enable to `Me` -> `App Settings` -> `Debug settings` -> `QuickStartForExistingUsersV2FeatureConfig`
3. Restart the app.
4. Logout/login and select "Show me around" when the quick start prompt is shown.
5. Select an existing site
6. ✅ Notice that the new `Quick Start` card with `Get to know the WordPress app`  task type is shown. 
7. Logout/login and select "Show me around" when the quick start prompt is shown.
8. Create a new site.
9. ✅ Notice that the old `Quick Start` card with `Customize` and `Grow` task types is shown.

#### Test.4 Site Picker - `QuickStartForExistingUsersV2FeatureConfig` enabled - Displays old `Quick Start` card when site is created.

1. Continue from Test.3.
2. Create a new site using the site picker.
3. ✅ Notice that the old Quick Start card with `Customize` and `Grow` task types is shown

#### Test.5 New Quick Start card - Common Tasks

1. ✅ Notice that the new Quick Start card with “Get to know the WordPress app”  task type currently has only three tasks that are common to the old Quick Start card: `check_stats`, `view_site`, `follow_site`.
2. Run tests in [wpandroid-quick-start-testing-instructions.txt](https://github.com/wordpress-mobile/WordPress-Android/files/8246096/wpandroid-quick-start-testing-instructions.txt).
3. ✅ Notice that all tests pass.

## Merge Instructions

1. Make sure `FluxC` PR is merged to trunk ✅ 
2. Update `FluxC` version in build.gradle ✅ 
3. Remove `Not Ready for Merge` label ✅ 
4. Merge the PR

## Regression Notes
1. Potential unintended areas of impact
Wrong `Quick Start Card` is shown based on whether site selected or created from login epilogue.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
See `To test` section

3. What automated tests I added (or what prevented me from doing so)
- Added unit tests for start quick start based on existing/ new site. - e6e9563
- Added unit tests for `check_stats`, `view_site` tasks for the new Quick Start card. - 6651903
- Yet to add tests for `follow_site` task.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by jostnes** [comment]
Did some testing on this build and overall the feature flag looks to be working as expected as per the above scenarios. I have two observations to share from testing:
1. My test phone is Oppo Reno 6Z (Google says that the screen size is 6.43-inch) but on this phone, the text on the new card seems to be too long, overlapping with the "Done" icon:
qsv2

2. When I restarted the app after updating the feature flag, I still get the previous QS version. It will only be updated to the correct one after I logout/relogin. I'm imagining the case where, when this feature is turned on for users, users who are already seeing the older QS should continue seeing it, is that correct? And only users who go through the Show me Around prompt will get to see the new version?

File: WordPress/src/main/java/org/wordpress/android/ui/quickstart/QuickStartType.kt [hunk 1]
```diff
@@ -50,4 +50,17 @@ sealed class QuickStartType(val tasks: List<QuickStartTask>) {
         return quickStartStore.getDoneCount(siteLocalId) >= tasks
                 .filter { it.taskType != QuickStartTaskType.UNKNOWN }.size
     }
+
+    companion object {
+        private const val NEW_SITE = "new-site"
+        private const val EXISTING_SITE = "existing-site"
+
+        fun fromString(label: String): QuickStartType {
```
Comment (AjeshRPai) [inline WordPress/src/main/java/org/wordpress/android/ui/quickstart/QuickStartType.kt]: Nitpick:  I think we can rename the function to `fromLabel`, WDYT? 🤔

**Review by AjeshRPai — APPROVED** [review]
@ashiagr I have tested the code changes according to the test instructions. All the changes are good. 👍🏼 🙌🏼 

I have left a nitpick for you to consider, as that is non blocking I approving the PR. Please feel free to merge once you have addressed it.

File: WordPress/src/main/java/org/wordpress/android/ui/quickstart/QuickStartType.kt (hunk 1 above)
Comment (ashiagr) [reply WordPress/src/main/java/org/wordpress/android/ui/quickstart/QuickStartType.kt]: Renamed in a97be94

**Comment by ashiagr** [comment]
Thanks, @AjeshRPai 

I've addressed your comment in a97be94.

Thanks, @jostnes. Please see my responses below:

> 1. My test phone is Oppo Reno 6Z (Google says that the screen size is 6.43-inch) but on this phone, the text on the new card seems to be too long, overlapping with the "Done" icon:

I've created an issue https://github.com/wordpress-mobile/WordPress-Android/issues/16539 to track it.

> 2. When I restarted the app after updating the feature flag, I still get the previous QS version. It will only be updated to the correct one after I logout/relogin. I'm imagining the case where, when this feature is turned on for users, users who are already seeing the older QS should continue seeing it, is that correct? And only users who go through the Show me Around prompt will get to see the new version?

This is correct. For more details, you can refer to this internal discussion: p1649849406838359-slack-C027K4MNPGQ

Since all comments are non-blocking, I'll go ahead and merge the PR once CI completes.

**Comment by osullivanchris** [comment]
Took a look at the build and all working as expected so far 👍 😄

**Pull request merged**